src/test/java/org/jvnet/hudson/test/JenkinsRuleTest.java

```diff
@@ -127,6 +142,206 @@ public void testTokenHelperMethods() throws Exception {
+                             webClient.withBasicApiToken(admin));
+        assertEquals(response.getStatusCode(), 200);
+        assertTrue(response.getContentAsString().contains("Jenkins is the way ! - UPDATED"));
+
+    }
+
+    @TestExtension
+    public static class JsonAPIForTests implements RootAction {
+
+        @CheckForNull
+        @Override
+        public String getIconFileName() {
+            return null;
+        }
+
+        @CheckForNull
+        @Override
+        public String getDisplayName() {
+            return null;
+        }
+
+        @Override
+        public String getUrlName() {
+            return "testing-cli";
+        }
+
+        @GET
+        @WebMethod(name = "getMyJsonObject")
+        public HttpResponse getMyJsonObject() {
+            JSONObject response = JSONObject.fromObject(new MyJsonObject("I am JenkinsRule"));
+            throw new JsonHttpResponse(response, 200);
+        }
+
+        @GET
+        @WebMethod(name = "getWithParameters")
+        public HttpResponse getWithParameters(@QueryParameter(required = true) String paramValue) {
+            assertNotNull(paramValue);
+            JSONObject response = JSONObject.fromObject(new MyJsonObject("I am JenkinsRule " + paramValue));
+            return new JsonHttpResponse(response, 200);
+        }
+
+        @GET
+        @WebMethod(name = "getError500")
+        public JsonHttpResponse getError500() {
+            JsonHttpResponse error500 = new JsonHttpResponse(
+                    JSONObject.fromObject(new MyJsonObject("You got an error 500")), 500);
+            throw error500;
+        }
+
+        @PUT
+        @WebMethod(name = "update")
+        public JsonHttpResponse update(@JsonBody MyJsonObject body) {
+            body.setMessage(body.getMessage()+" - UPDATED");
+            JSONObject response = JSONObject.fromObject(body);
+            return new JsonHttpResponse(response, 200);
+        }
+
+        @PUT
+        @WebMethod(name = "updateFailure")
+        public JsonHttpResponse updateFailure(@JsonBody MyJsonObject body) {
+            body.setMessage(body.getMessage()+" - NOT UPDATED");
+            JsonHttpResponse error500 = new JsonHttpResponse(
+                    JSONObject.fromObject(body), 500);
+            throw error500;
+        }
+
+        @POST
+        @WebMethod(name = "create")
+        public JsonHttpResponse create(@JsonBody MyJsonObject body) {
+            body.setMessage(body.getMessage()+" - CREATED");
+            JSONObject response = JSONObject.fromObject(body);
+            return new JsonHttpResponse(response, 200);
+        }
+
+        @POST
+        @WebMethod(name = "createFailure")
+        public JsonHttpResponse createFailure(@JsonBody MyJsonObject body) {
+            body.setMessage(body.getMessage()+" - NOT CREATED");
+            JSONObject response = JSONObject.fromObject(body);
+            return new JsonHttpResponse(response, 500);
```

Comment:
Just curious why this method returns the response and the other failure methods throw it.

Reply:
it's technically having the same behavior, but I prefer to throw when it's an exception, thanks for seeing it

Reply:
done.